net/dns/resolver/forwarder.go

```diff
@@ -58,10 +59,20 @@ func truncatedFlagSet(pkt []byte) bool {
 }
 
 const (
-	// dohTransportTimeout is how long to keep idle HTTP
-	// connections open to DNS-over-HTTPs servers. This is pretty
-	// arbitrary.
-	dohTransportTimeout = 30 * time.Second
+	// dohIdleConnTimeout is how long to keep idle HTTP connections
+	// open to DNS-over-HTTPS servers. 10 seconds is a sensible
+	// default, as it's long enough to handle a burst of queries
+	// coming in a row, but short enough to not keep idle connections
+	// open for too long. In theory, idle connections could be kept
+	// open for a long time without any battery impact as no traffic
+	// is supposed to be flowing on them.
+	// However, in practice, DoH servers will send keepalives (e.g.
+	// NextDNS sends them every ~10s). Handling these keepalives
```

Comment:
"them" ... what type of keep-alives? TCP layer? Or http-layer?

Reply:
TCP:

<img width="1609" alt="Screenshot 2024-10-02 at 09 17 30" src="https://github.com/user-attachments/assets/8379f76a-6546-4ff1-8717-b5731b239620">

Reply:
Comment updated.